yorkie/src/main/kotlin/dev/yorkie/document/crdt/RHTPQMap.kt

```diff
@@ -0,0 +1,161 @@
+package dev.yorkie.document.crdt
+
+import android.util.Log
+import dev.yorkie.document.time.TimeTicket
+import dev.yorkie.util.MaxPriorityQueue
+import dev.yorkie.util.PQNode
+
+
+/**
+ * RHTPQMap is replicated hash table with priority queue by creation time.
+ *
+ * @internal
+ */
+internal class RHTPQMap {
+    private val logTag = "RHTPQMap"
+
+    private val elementQueueMapByKey: MutableMap<String, MaxPriorityQueue<PQNode<TimeTicket, CrdtElement>>> =
+        mutableMapOf()
+    private val nodeMapByCreatedAt: MutableMap<String, RHTPQMapNode> = mutableMapOf()
+
+    /**
+     * `set` sets the value of the given key.
+     *
+     * @param key to get queue object from [elementQueueMapByKey]
+     * @param value The CrdtElement object to set [elementQueueMapByKey]
+     * @return CrdtElement if the object exist in [elementQueueMapByKey] by same key, it will return.
+     */
+    fun set(key: String, value: CrdtElement): CrdtElement? {
+        var removed: CrdtElement? = null
+        val queue = elementQueueMapByKey[key]
+        if (queue != null && queue.size > 0) {
+            val node = queue.peek() as RHTPQMapNode
+            if (!node.isRemoved() && node.remove(value.createdAt)) {
+                removed = node.value
+            }
+        }
+        setInternal(key, value)
+        return removed
+    }
+
+    /**
+     * `setInternal` sets the value of the given key internally.
+     */
+    private fun setInternal(key: String, value: CrdtElement) {
```

Comment:
Access modifier was public in `yorkie-js-sdk`, but I didn't find any caller from outside. 
So I wrote this function using `private`, If it's incorrect, please let me know.

Reply:
Thanks for the good comments. I'll change it to private in the JS SDK as well.